geoutils/georaster/raster.py

```diff
@@ -527,13 +527,15 @@ def __str__(self) -> str:
 
     def __eq__(self, other: object) -> bool:
         """Check if a Raster's data and georeferencing is equal to another."""
-        from geoutils.misc import array_equal
 
         if not isinstance(other, type(self)):  # TODO: Possibly add equals to SatelliteImage?
             return NotImplemented
         return all(
             [
-                array_equal(self.data, other.data, equal_nan=True),
+                np.array_equal(self.data.data, other.data.data, equal_nan=True),
```

Comment:
Sorry if I'm debating things that we maybe already discussed...
Should we not just compare the valid values here? I agree that it's important to compare the masks, but I feel that comparing the values of `self.data.data` that are masked could lead to a lot of confusions for most users...

Reply:
Actually, I think the \_\_eq\_\_ feature will be the more intuitive for users, as they just have to run `rst1 == rst2`. So I would make sure this one does not have counter-intuitive tests, like checking masked values. We could have a function called `strictly_equal` for that, that we would mostly only use for tests.

Reply:
Issue opened #316